### PR TITLE
Sync parser with latest mobile improvements

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add latest changes from mobile, which include:
- Better error handling
- Show errors on suggestions
- Add flag name to suggestion in text fields

#### Ticket Link
None

### CODE REVIEW

File: components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts [hunk 1]
```diff
@@ -765,7 +760,7 @@ describe('AppCommandParser', () => {
                     Description: 'The Jira issue summary',
                     Hint: '',
                     IconData: 'Create icon',
-                    Suggestion: 'Sum',
+                    Suggestion: 'summary: `Sum`',
```
Comment (mickmister) [inline components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:763]: Should we maybe use `ModalLabel` here if it exists? It's more of a "proper" label for the field

**Review by mickmister — APPROVED** [review]

File: components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts (hunk 1 above)
Comment (larkox) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:763]: I have mixed feelings, because it may create some discrepancies. A field that is:
```
{
    location: 'ticket-name'
    label: 'ticket-name'
    modalLabel: 'Ticket Name`
}
```
will have you write ticket-name, and then show `Ticket Name: ""`. I don't know. In this case it is pretty obvious, but I wonder if there is any use case where this may become more confusing. @abhijit-singh thoughts?

**Review by catalintomai — APPROVED** [review]

**Review by mickmister — APPROVED** [review]

File: components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts (hunk 1 above)
Comment (abhijit-singh) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:763]: @larkox Going by the example given here, I agree that we should use `ModalLabel`. If we come across a case in the future where it isn't as obvious let's flag it and revisit this if needed.

File: components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts [hunk 2]
```diff
@@ -443,11 +443,6 @@ describe('AppCommandParser', () => {
     });
 
     describe('getSuggestions', () => {
-        test('just the app command', async () => {
-            const suggestions = await parser.getSuggestions('/jira');
-            expect(suggestions).toEqual([]);
-        });
-
```
Comment (mickmister) [inline components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:450]: Why was this removed?

Comment (larkox) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:450]: Because this is not handle by `parser.getSuggestions` but by `parser.getBaseSuggestions` and with the latest changes the error this generated was being surfaced.

Comment (larkox) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:450]: I have just discovered this is not true. I will recover this test and fix the issue.

Comment (larkox) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.test.ts:450]: After looking it further, I was right at the beginning. The getSuggestions should never be called before the base command is complete. Before that we usually call getsuggestionbase. The moment we add a space, it starts using the getsuggestions function.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested all command parser help
- Help options added all seem to be working as expected
- This works consistently with previous mobile PR 

LGTM!
